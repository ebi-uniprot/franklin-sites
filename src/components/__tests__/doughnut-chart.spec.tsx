import { render, screen } from '@testing-library/react';

import DoughnutChart from '../doughnut-chart';

describe('DoughnutChart component', () => {
  it('should render low percent', () => {
    const { asFragment } = render(
      <DoughnutChart percent={40}>
        <span>Some content</span>
      </DoughnutChart>
    );
    // TODO: this should be a visual snapshot eventually
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render high percent', () => {
    const { asFragment } = render(
      <DoughnutChart percent={60}>
        <span>Some content</span>
      </DoughnutChart>
    );
    // TODO: this should be a visual snapshot eventually
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without content', () => {
    const { asFragment } = render(<DoughnutChart percent={60} />);
    expect(screen.queryByText('60%')).not.toBeNull();
    // TODO: this should be a visual snapshot eventually
    expect(asFragment()).toMatchSnapshot();
  });
});
