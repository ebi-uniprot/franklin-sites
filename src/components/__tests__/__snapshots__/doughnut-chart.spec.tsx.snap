// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoughnutChart component should render high percent 1`] = `
<DocumentFragment>
  <span
    class="doughnut-chart--medium colour-platinum"
  >
    <span
      class="doughnut-chart--medium__left-wrap"
    >
      <span
        class="doughnut-chart--medium__left-wrap__loader colour-sea-blue"
        style="transform: rotate(36deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__right-wrap"
    >
      <span
        class="doughnut-chart--medium__right-wrap__loader colour-sea-blue"
        style="transform: rotate(180deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__inner-circle"
    >
      <span>
        Some content
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`DoughnutChart component should render low percent 1`] = `
<DocumentFragment>
  <span
    class="doughnut-chart--medium colour-platinum"
  >
    <span
      class="doughnut-chart--medium__left-wrap"
    >
      <span
        class="doughnut-chart--medium__left-wrap__loader colour-sea-blue"
        style="transform: rotate(0deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__right-wrap"
    >
      <span
        class="doughnut-chart--medium__right-wrap__loader colour-sea-blue"
        style="transform: rotate(144deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__inner-circle"
    >
      <span>
        Some content
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`DoughnutChart component should render without content 1`] = `
<DocumentFragment>
  <span
    class="doughnut-chart--medium colour-platinum"
  >
    <span
      class="doughnut-chart--medium__left-wrap"
    >
      <span
        class="doughnut-chart--medium__left-wrap__loader colour-sea-blue"
        style="transform: rotate(36deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__right-wrap"
    >
      <span
        class="doughnut-chart--medium__right-wrap__loader colour-sea-blue"
        style="transform: rotate(180deg);"
      />
    </span>
    <span
      class="doughnut-chart--medium__inner-circle"
    >
      <span>
        60%
      </span>
    </span>
  </span>
</DocumentFragment>
`;
